manager/integration/tests/common.py

```diff
@@ -3337,3 +3339,36 @@ def settings_reset():
 
     client = get_longhorn_api_client()
     reset_settings(client)
+
+
+def wait_for_instance_manager_desire_state(client, core_api, im_name,
+                                           state, desire=True):
+    time.sleep(2 * RETRY_INTERVAL)
```

Comment:
This is kind of random. Why do we need this?

Reply:
Giving sometime before we check the status. Say if we check the status immediately after we update the setting, I am not sure if the status were still in the previous states.

Reply:
For this function, if the caller is not sure the state is good to check for the `wait_for`, the caller should take steps to ensure the state is good to check before calling the function. In this case, at least move the `sleep` into the caller.

Normally we shouldn't have any random `sleep` call since you won't know how long is enough given a certain environment. That's what introduced race condition. In this case, we can check the previous pods' UUID before and after, to make sure the change is happening.